File: Examples/test-suite/ruby/director_rev_len_str_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#

require 'swig_assert'

require 'director_rev_len_str'

class DirectorBinaryStringCallback < Director_rev_len_str::Callback
  def run(dataBufferAA)
    ret = 0
    unless dataBufferAA.nil?
        dataBufferAA.each_char do |c|
            ret += c.ord * 2
        end
    end
    return ret
  end
end

caller = Director_rev_len_str::Caller.new(DirectorBinaryStringCallback.new)
sum = caller.call()

raise RuntimeError if sum != 9*2*8

File: Examples/test-suite/python/director_rev_len_str_runme.py
import director_rev_len_str

class DirectorBinaryStringCallback(director_rev_len_str.Callback):
    def run(self, dataBufferAA):
        ret = 0
        for c in dataBufferAA:
            ret += ord(c) * 2
        return ret


callback = DirectorBinaryStringCallback()
caller = director_rev_len_str.Caller(callback)
sum = caller.call()

if sum != 9*2*8:
    raise RuntimeError("Unexpected sum: {}".format(sum))
